src/wheatley.ts

```diff
@@ -427,30 +434,100 @@ export class Wheatley extends EventEmitter {
                     }
                 }
                 // TODO: Handle unexpected input?
-                // NOTE: For now only able to take text input
-                assert(
-                    [...command.options.values()].every(option => (option.type as any) == "string"),
-                    "unhandled option type",
-                );
-                const parts = string_split(
-                    message.content.substring(match[0].length).trim(),
-                    " ",
-                    command.options.size,
-                );
+                // NOTE: For now only able to take text and user input
+                // TODO: Handle `required`
+                let command_body = message.content.substring(match[0].length).trim();
+                const command_options: unknown[] = [];
                 for (const [i, option] of [...command.options.values()].entries()) {
-                    if (i >= parts.length && option.required) {
-                        await command_obj.reply({
-                            embeds: [
-                                create_basic_embed(
-                                    undefined,
-                                    colors.red,
-                                    `Required argument "${option.title}" not found`,
-                                ),
-                            ],
-                        });
-                        return;
+                    if (option.type == "string") {
+                        if (option.regex) {
+                            const match = command_body.match(option.regex);
+                            if (match) {
+                                command_options.push(match[0]);
+                                command_body = command_body.slice(match[0].length).trim();
+                            } else {
+                                await command_obj.reply({
```

Comment:
For the sake of readability, since the pattern of `await make error embed;` pops up a bunch of times, I think there should be a local function here to avoid code duplication:
```js
const replyWithError(msg: string) => command_obj.reply({
    embeds: [
        create_basic_embed(undefined, colors.red, msg),
    ],
});

// ...
await replyWithError(`Required argument "${option.title}" not found`);
return;
```